sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -83,6 +83,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
         BooleanSimplification,
         SimplifyConditionals,
         RemoveDispensableExpressions,
+        ReplaceNullWithFalse,
```

Comment:
Logically, `ReplaceNullWithFalse` can use the result of `SimplifyBinaryComparison`. How about postponing this after `SimplifyBinaryComparison`? In other words, switch `ReplaceNullWithFalse` and `SimplifyBinaryComparison`?